src/charm.py

```diff
@@ -431,25 +431,36 @@ def k8s_auth(self):
 
         self._authed = True
 
+    def _core_v1_api(self) -> Any:
+        """Use the v1 k8s API."""
+        self.k8s_auth()
+        return kubernetes.client.CoreV1Api()
+
+    def _networking_v1_api(self) -> Any:
```

Comment:
Wouldn't the type be `NetworkingV1Api`? I guess for the functions add/ changed in the PR it would be good to write type hints that aren't `Any`. For the rest, I can see an argument for that it should be done in a follow up PR

Reply:
Yes, it would be nice to have a closer look at this together with the plugin for Kubernetes types + mypy.